.scalafix.conf

```diff
@@ -40,6 +40,29 @@ Disable.ifSynthetic = [
   }
 ]
 
+DisableSyntax {
```

Comment:
Unfortunately not all the rules are explained in the documentation :( in which case I am not very happy about having them enabled. For example `noImplicitConversion` is not documented - does that mean that even extension methods via `implicit class` are not allowed? `noFinalVal` is documented but very poorly so I don't understand the reasoning why it shouldn't be allowed.  Why `noImplicitObject`? That seems like something we might want to have in our code. I also don't understand the contravariant and covariant rules... otherwise the rest seems reasonable to me.

Reply:
> `noImplicitConversion` is not documented - does that mean that even extension methods via implicit class are not allowed

I think I understand where you're coming from, and I haven't tested it, but I don't think that's the case. In Scala community, implicit conversions and implicit classes are considered different things, so I expect scalafix/DisableSyntax to be aligned with this common understanding.
If it turns out that this rule prevents us from creating/using extension methods, I'm all for disabling it.

> `noFinalVal`

I think the [bug report](https://github.com/sbt/zinc/issues/227) explains it -- it's a workaround a bug in Scala compilation toolchain.

> `noImplicitObject`

[This is the explanation that I've found](https://github.com/scalacenter/scalafix/blob/master/scalafix-rules/src/main/scala/scalafix/internal/rule/DisableSyntax.scala#L214).  [Here's probably what's an example of this problematic behavior](https://stackoverflow.com/questions/20380800/scala-implicits-resolution-mechanism-is-declaration-order-dependent)

> Implicits without explicit return types are not eligible in implicit searches higher up in the same file. Without this rule, the typechecker runs into cycles very quickly. Implicit object cannot have an explicit type ascribed; my recommendation is to avoid them in favour if implicit vals. – retronym (_a Scala compiler developer_)

Reply:
@sideeffffect Please check that `noImplicitConversion` behaves the expected way.

@hanny24 What do you think about `noFinalVal`? I've never encountered or heard about that issue and it seems strange to me to ban some regular Scala language feature just because incremental compilation might not work properly on your computer. I would remove this rule.

`noImplicitObject` - even though I've also never heard or encountered it I guess we can have this rule to prevent possible unnecessary issues.

Reply:
the Zinc issue is actually explained very very well! The first link in the issue in the documentation https://github.com/sbt/sbt/issues/1543

Reply:
Yes, that comment explains it well but the impact to the regular Scala developer is quite abstract. I've clicked through the different issues and some of them are closed and the fact that this isn't resolved even after many years indicates that this is not critical issue encoutered often. I am defintely against having it in the ruleset.

Reply:
@hanny24 @tomasherman if you have a strong desire to have `noFinalVal` rule enforced, now's the time to speak up :) otherwise I'll be taking it out some time in the afternoon

Reply:
I actually don't know what does `noFinalVal` do...

Reply:
not really - i don't think we should limit people trying to contribute unless there is really really strong case for it

Reply:
aaand... it's gone :smiley:

Reply:
@sideeffffect From the discussion above I think we're still missing a confirmation that `noImplicitConversion` really bans only what we want and not implicits classes for example.